yandex_contest/binary_sum.py

```diff
@@ -1,44 +1,35 @@
-def convert(string):
-    list_1 = []
-    list_1[:0] = string
-    list_x = [int(i) for i in list_1]
+def convert_to_list(string):
+    list_x = [int(i) for i in string]
     return list_x
 
+
 def main():
     num_1 = input()
-    list_1 = convert(num_1)
+    list_1 = convert_to_list(num_1)
 
     num_2 = input()
-    list_2 = convert(num_2)
+    list_2 = convert_to_list(num_2)
+
+    reversed_num_1 = list(reversed(list_1))
+    reversed_num_2 = list(reversed(list_2))
+
+    if len(reversed_num_1) > len(reversed_num_2):
+        temp = reversed_num_1
+        reversed_num_1 = reversed_num_2
+        reversed_num_2 = temp
+
+    for _ in range(len(reversed_num_2) - len(reversed_num_1)):
+        reversed_num_1.append(0)
```

Comment:
There is a nice trick in python. 
Basically, you need to append N zeroes to the array. You can either do it with a for loop as you did or you can use [`.extend()`](https://docs.python.org/3/tutorial/datastructures.html) method of the list and a `[0] * N` syntax to generate a list of size N full of zeros.

So the code will look like 
```suggestion
    reversed_num_1.extend([0] * (len(reversed_num_2) - len(reversed_num_1)))
```

Reply:
But in general, we don't even need to do this! These operations make us use more memory by appending zeroes we don't really need.
What if in the `for`-loop below we will compare the index with a size of the smallest array (which we now know is always `list_1`!) and if the index is outside we will just use 0 instead of the element?